lib/osut/utils.rb

```diff
@@ -2098,12 +2116,30 @@ def addSubs(model = nil, s = nil, subs = [], clear = false, bfr = 0.005)
           log(WRN, "Reset count (ratio) to 1 (#{mth})")
         end
 
-        area  = gross * sub[:ratio] # sub m2, including (optional) frames
+        area  = s.grossArea * sub[:ratio] # sub m2, including (optional) frames
         w     = area / h
         width = w - frames
         x0    = centre - w/2
         xf    = centre + w/2
 
+        if sub.key?(:l_buffer)
+          if sub.key?(:centreline)
+            log(WRN, "Skip #{id} left buffer (vs centreline) (#{mth})")
+          else
+            x0     = sub[:l_buffer] - frame
+            xf     = x0 + w
+            centre = x0 + w/2
+          end
+        elsif sub.key?(:r_buffer)
+          if sub.key?(:centreline)
+            log(WRN, "Skip #{id} right buffer (vs centreline) (#{mth})")
+          else
+            xf     = max_x - sub[:r_buffer] + frame
+            x0     = xf - w
+            centre = x0 + w/2
+          end
+        end
```

Comment:
Attribute **:centreline** has precedence over either **left** or **right** buffer attributes. If both **left** and **right** buffers are requested by the user, **left** is favoured over **right**.